### DISCUSSION TITLE
v0.90.12

### DISCUSSION THREAD

**Discussion by eliandoran** [TriliumNext/Trilium#6025] [Releases]
💡 Key highlights
-----------------

*   Now it is possible to have a fixed toolbar for editing text notes instead of the standard floating one, see [Classic editor for text notes (with fixed toolbar)](https://github.com/TriliumNext/Notes/pull/571) for more information.
*   Syntax highlight in code blocks in editable and read-only text notes (via Highlight.js), heavily based on [antoniotejada](https://github.com/antoniotejada)'s [Trilium-SyntaxHighlightWidget](https://github.com/antoniotejada/Trilium-SyntaxHighlightWidget).

Changes since the last beta release
-----------------------------------

### 🐞 Bugfixes

*   [Shell (Bash) Syntax Highlight](https://github.com/TriliumNext/Notes/issues/583)
*   [0 Max content width causes editor issues](https://github.com/TriliumNext/Notes/issues/593)
*   [Share icon not visible in note icon picker](https://github.com/TriliumNext/Notes/issues/603)

### ✨ Improvements

*   Add find & replace button to fixed toolbar
*   [Add more link protocol support](https://github.com/TriliumNext/Notes/pull/589) by @SiriusXT
*   [Add a text replacement feature to the find\_widget](https://github.com/TriliumNext/Notes/pull/576) by @SiriusXT
*   [Replace the editor type combo box with radio buttons](https://github.com/TriliumNext/Notes/pull/587) by @adoriandoran
*   [Add box icons to note menu & attachments](https://github.com/TriliumNext/Notes/pull/609) by @SiriusXT
*   [Improve the "Existing backups" section](https://github.com/TriliumNext/Notes/pull/615) by @adoriandoran
*   [Trigger full text search when Ctrl + Enter is pressed in note autocomplete](https://github.com/TriliumNext/Notes/pull/585) by @SiriusXT
*   [Improve context menus](https://github.com/TriliumNext/Notes/pull/618) (+ [Replace (?) with information icon](https://github.com/TriliumNext/Notes/issues/613)) by @adoriandoran
*   [Add Options Launcher](https://github.com/TriliumNext/Notes/issues/619)
*   [Change note clone icon indicator from star to link icon](https://github.com/TriliumNext/Notes/issues/565)
*   [Add "Reopen last tab" and "Copy tab to new window" to tab management](https://github.com/TriliumNext/Notes/pull/651) by @SiriusXT

### 🌍 Internationalization

*   Improved Spanish translations by @hasecilu
*   Improved German translations by @j13055
*   [Improve note revision wording and consistency](https://github.com/TriliumNext/Notes/pull/612) by @meichthys 

### ⬆️ Library updates and technical improvements

*   [Add renovate GitHub Action and JSON config](https://github.com/TriliumNext/Notes/pull/607) by @perfectra1n 

From `v0.90.11-beta`
--------------------

### 🐞 Bugfixes

*   [Double input after spaces on Android](https://github.com/TriliumNext/Notes/issues/568)
*   [Hyperlinks with custom protocols not working](https://github.com/TriliumNext/Notes/issues/122) by @SiriusXT
*   [Internal Link: Two Tooltips are displayed](https://github.com/TriliumNext/Notes/issues/525)
*   Fixed loading of syntax highlighting themes for server and Docker builds.

### ✨ Improvements

*   [Insert inline images in text notes](https://github.com/TriliumNext/Notes/issues/531) (also grouped the image alignment buttons in subcategories since there were too many of them).
*   [Explicitly manage the "latest" tag, and have it point to the same tag as "stable"](https://github.com/TriliumNext/Notes/pull/545) by @perfectra1n
*   [Make attachments open in a new tab/browser](https://github.com/TriliumNext/Notes/pull/559) by @SiriusXT
*   [mouse scroll wheel direction for zoom level](https://github.com/TriliumNext/Notes/pull/555) by @rom1dep
*   Apply syntax highlight to preview of code notes as well.
*   [More reliably check for version updates](https://github.com/TriliumNext/Notes/pull/574) by @perfectra1n
*   README improvements by @perfectra1n

### 🌍 Internationalization

*   Preliminary support for the German language by @j13055 
*   Improved French translations by @Potjoe-97
*   Improved Spanish translations by @hasecilu

### ⬆️ Library updates

*   mind-elixir: 4.2.4 -> 4.3.1
*   client: force-graph: 1.45.0 -> 1.46.0

From `v0.90.10-beta`
--------------------

### 🐞 Bugfixes

*   Code note overlapping with note list.
*   Fix error when running a script in 0.90.9-beta.
*   Font section in Appearance settings blinking when settings were being updated.
*   [Crop fileName and prevent cutting into the extension](https://github.com/TriliumNext/Notes/pull/541) / Canvas note exported to html can't be view due to title name is 13-15 digits long by @SiriusXT
*   Duplicate note count in delete relation modal.
*   Docnotes (e.g. launcher configuration descriptions) not showing on Electron/desktop builds.

### ✨ Improvements

*   Adjustable word-wrap for code blocks in text notes.
*   Adjustable theme for code blocks in text notes, plus slightly improved design.
*   The application should now be more highly tolerant to errors caused by custom widgets and scripts, not causing the UI to also crash (see [#511](https://github.com/TriliumNext/Notes/issues/511)).
*   [Close tabs to the right](https://github.com/TriliumNext/Notes/pull/542) by @SiriusXT
*   [Automatically trigger autocomplete on focus](https://github.com/TriliumNext/Notes/pull/534) by @SiriusXT

### 🌍 Internationalization

*   Translate weekday and month names for daily notes.
*   Update server translations when switching language.
*   Translate search note prefix.
*   Translate sync test messages.

### ⬆️ Library updates

*   mind-elixir: 4.2.3 → 4.2.4
*   mermaid: 11.3.0 → 11.4.0
*   i18next: 23.16.2 -> 23.16.4
*   and other internal development tooling updates.

From `v0.90.9-beta`
-------------------

### 🐞 Bugfixes

*   [Promoted boolean attributes no long showing checkbox when the value is true](https://github.com/TriliumNext/Notes/issues/503)
*   [Import of .htm keeps htm in title while html does not](https://github.com/TriliumNext/Notes/issues/500)

### ✨ Improvements

*   [Syntax highlighting in read-only mode code notes](https://github.com/TriliumNext/Notes/issues/504) (also for “View source”).
*   [Server 404 logs are now one-line instead of a full stack trace](https://github.com/TriliumNext/Notes/issues/505)

### 🌍 Internationalization

*   Translated the toast notifications, errors and confirmation boxes.
*   Translated the launcher context menu.

### ⬆️ Library updates

*   i18next: 23.16.0 -> 23.16.2
*   mind-elixir: 4.2.0 -> 4.2.2
*   vanilla-js-wheel-zoom: 9.0.2 -> 9.0.4
*   cookie-parser: 1.4.6 → 1.4.7
*   image-type: 4.1.0 → 5.2.0
*   express, express-rate-limit, express-session updated to latest.
*   cookie-parser: 1.4.6 → 1.4.7
*   marked: 14.1.2 → 14.1.3
*   sanitize-html: 2.13.0 → 2.13.1
*   force-graph: 1.43.5 → 1.45.0
*   and others (jasmine, debounce)

<hr /><em>This discussion was created from the release <a href='https://github.com/TriliumNext/Notes/releases/tag/v0.90.12'>v0.90.12</a>.</em>

**Comment by doobes** [comment]
Great news,

To update my server, should I change:

image: 'triliumnext/notes:v0.90.6-beta'

to

image: 'triliumnext/notes:v0.90.12'

Thanks!

**Reply by meichthys** [reply]
That should work or 'triliumnext/notes:latest' if you don't want to pin to a specific version. This would keep you on the latest stable image.

**Reply by doobes** [reply]
Awesome.  Next silly question. 

I downloaded the latest deb for linux and installed via sudo dpkg -i TriliumNextNotes-v0.90.12-linux-x64.deb.

Opening TNN, it tells me it's still v0.90.4.

Advice?

**Reply by perfectra1n** [reply]
Then you have multiple Notes binaries installed on your computer, and you're executing the old one. Did `dpkg` note that it was upgrading from `v0.90.4` to `v0.90.12`, or just installing?

**Reply by meichthys** [reply]
How are you launching TNN?
Could it be there is an older installation that is being run?

**Reply by doobes** [reply]
Okay,

A chalk and cheese issue.

Previously, I had simply unpacked TNN into a folder and ran using the appropriate sh file. 

Then I bone headedly decided to install it using dpkg.

Dumb and dumber.

Cheers

**Comment by meichthys** [comment]
'dpkg --list | less' to see what's installed, then 'apt remove package_name' should help.

**Comment by harish-shivaram** [comment]
I am trying to install the MAC OS ARM64 version on my M1 Macbook air. I have never installed TNN on my macbook before.

When I try to launch the application, I get the error `"TriliumNext Notes" is damaged and can't be opened. You should move it to the Bin.`

Any help is appreciated.

**Reply by eliandoran** [reply]
Hi @harish-shivaram , this happens because our application is not yet notarized/signed by Apple (we would need to buy an Apple developer's certificate).

You can bypass it by following the instructions here: https://github.com/TriliumNext/Notes?tab=readme-ov-file#macos.
